### ISSUE TITLE
function to select ideal_helix either with get_atomic_residues or get_ideal_helix

### ISSUE THREAD

**Issue by ichem001** [salilab/pmi#186]
When creating a multiple subunits system with structures, ideal_helix, and beads:
is it possible to assign ideal_helix as atomic_residues (it is not until you build hierarchy) or have its own selector?

**Comment by cgreenberg** [comment]
What do you mean "assign ideal_helix as atomic_residues"? When you call this:

`mol.add_representation(mol[0:100], resolutions = [1], ideal_helix=True)`

 that creates resolution 1 beads in the shape of a helix. Do you mean you want to create side chains? You can always select those beads with `IMP.atom.Selection`

**Comment by ichem001** [comment]
I mean create a selector for ideal_helix. 
The current workaround is to create a list with the fragment corresponding to ideal helices:
IH = [mol[X1:Y1], mol[X2, Y2],...], which works fine as long as you have only one of those list. 
Working on bigger complexes (more than a few components) it gets tricky. 
So adding a function that return the residues with ideal_helix representation would be useful.

For example this piece of code should illustrate what I mean:
I added representation for my structured part ("atomic") and my ideal helices (resolutions=[1]). 
When you want to create the beads for unstructured part: something like that would be useful
mol.add_representation(mol[:] - mol.get_atomic_residues() - **mol.get_ideal_helices()**, resolutions=[10], color=colors[n]) 

Otherwise you have the exception: "You have already added representation for XXX"

Is that clearer?

**Comment by shruthivis** [comment]
Agree with Ilan and think it will be nice to have such a function too for
the same reason (or annotate ideal helices as atomic_residues, making the
behavior of get_atomic_residues same before and after build). This function
may also be useful while creating dof rigid bodies for each helix.

**Comment by cgreenberg** [comment]
Oh, I see now. @ichem001 do you agree it would be sufficient to let the ideal helices be marked as "structured" so that they're returned with `get_atomic_residues()`?

**Comment by shruthivis** [comment]
Actually how about creating the separate function returning the list of
ideal_helices so that we can call this list while creating rigid bodies?
I prefer this because marking helices as" structured" is fine for the
purpose of creating flexible bead representations, however, it is not fine
for the purpose of setting up degrees of freedom. Because these ideal
helices and structured regions are usually created as separate rigid
bodies, it will be nice to have them as separate lists.

On Fri, Mar 11, 2016 at 4:33 PM, Charles Greenberg <notifications@github.com

**Comment by cgreenberg** [comment]
I guess I don't really know why it's needed. I mean, if you create the helices, you obviously know which residues they are, so you can just collect them and do set operations on them like you wanted. 

``` python
mol.add_representation(mol[a:b],ideal_helix=True)
mol.add_representation(mol[c:d],ideal_helix=True)
dof.create_rigid_body(mol[:] - (mol[a:b] | mol[c:d]))
```

My concern is if we start storing too many things in the molecule class then we just end up replicating functionality. E.g., why not also store all beads at resolution 1, 10, 100 in separate lists? At least with `get_atomic_residues()` it's useful because when you add structure it has lots of holes in it so it's hard to assemble from segments.

**Comment by cgreenberg** [comment]
Ok well if we add it, should helices also be mark as "atomic"?

**Comment by shruthivis** [comment]
I see. I can definitely live without the function. But it maybe good to
mark helices as atomic in any case just to make the behavior same before
and after build.

**Comment by ichem001** [comment]
I guess it is one way to do it.
Aesthetically, when you have many such ideal_helices per component, and many components, the script can get ugly really quickly.

**Comment by cgreenberg** [comment]
OK, will add `Molecule.get_ideal_helices()`. I suppose the best thing is to return a list of lists of contiguous TempResidues, corresponding to each helix within that molecule, yes?

**Comment by shruthivis** [comment]
Sounds good, thanks.

On Mon, Mar 14, 2016 at 2:21 PM, Charles Greenberg <notifications@github.com
